elasticdl_preprocessing/layers/log_round.py

```diff
@@ -0,0 +1,81 @@
+import tensorflow as tf
+from tensorflow.python.ops.ragged import ragged_functional_ops, ragged_tensor
+
+
+def log(x, base):
+    x = tf.cast(x, tf.float64)
+    numerator = tf.math.log(x)
+    denominator = tf.math.log(tf.constant(base, dtype=numerator.dtype))
+    return numerator / denominator
+
+
+class LogRound(tf.keras.layers.Layer):
+    """Cast a numeric feature into a discrete integer value by
+    `round(log(x))`.
+
+
+    Example :
+    ```python
+        layer = LogRound(num_bins=16, base=2)
+        inp = np.asarray([[1.2], [1.6], [0.2], [3.1], [100]])
+        layer(inp)
+        [[0], [1], [0], [2], [7]]
+    ```
+
+    Arguments:
+        num_bins: Range of inputs and outputs is `[0, num_bins)`.
+        **kwargs: Keyword arguments to construct a layer.
+
+    Input shape: A numeric `Tensor`, `SparseTensor` or `RaggedTensor` of shape
+        `[batch_size, d1, ..., dm]`
+
+    Output shape: An int64 tensor of shape `[batch_size, d1, ..., dm]`
+
+    """
+
+    def __init__(self, num_bins, default_value=0, base=None):
```

Comment:
I think we need a `offset` or `bias` parameter.

Reply:
Can you give a use case to explain why we need it?

Reply:
What if user pass `0` as input?

Reply:
If the input is zero or a minus, the output of `tf.math.log` is -inf. Then, `tf.where` will replace it with zero to output.
```python
values = tf.where(
            tf.logical_or(values < 0, values >= num_bins),
            x=tf.fill(dims=tf.shape(values), value=default_value),
            y=values,
        )
```
And I have added the test case with input `0` in the unit test.

Reply:
We can omit the bias parameter. If user needs, we can add it later.